pkg/plugins/errors.go

```diff
@@ -3,16 +3,24 @@ package plugins
 import "github.com/grafana/grafana/pkg/util/errutil"
 
 var (
+	errPluginNotRegisteredBase = errutil.NotFound("plugin.notRegistered",
+		errutil.WithPublicMessage("Plugin not registered"))
 	// ErrPluginNotRegistered error returned when a plugin is not registered.
-	ErrPluginNotRegistered = errutil.NotFound("plugin.notRegistered")
-	// ErrHealthCheckFailed error returned when a plugin health check failed.
-	ErrHealthCheckFailed = errutil.Internal("plugin.failedHealthCheck")
+	ErrPluginNotRegistered = errPluginNotRegisteredBase.Errorf("%w", errPluginNotRegisteredBase)
+
+	errPluginUnavailableBase = errutil.Internal("plugin.unavailable",
+		errutil.WithPublicMessage("Plugin unavailable"))
 	// ErrPluginUnavailable error returned when a plugin is unavailable.
-	ErrPluginUnavailable = errutil.Internal("plugin.unavailable")
+	ErrPluginUnavailable = errPluginUnavailableBase.Errorf("%w", errPluginUnavailableBase)
```

Comment:
This construction looks a bit funky and hard to see what the error message going to be? Based on below seems like it will create an empty error https://github.com/grafana/grafana/blob/9608da4fdff778decdb165ea127759fc67d30c5a/pkg/util/errutil/errors.go#L229-L231  

Maybe better to write `"plugin unavailable"`? And same for below errors. WDYT?

Reply:
not sure if you saw the last commit: https://github.com/grafana/grafana/pull/74928/commits/89c5c1daa62940c0c6989df304efe2199e4d6ed2

I changed the public errors, in this case it will return a "Plugin unavailable"

Reply:
I did. But it's not the same thing as the error message. Public message being returned in API error responses and the error message is normally only logged (if app_mode = production)